### ISSUE TITLE
`cargo crev fetch` does not work with ssh URLs

### ISSUE THREAD

**Issue by BatmanAoD** [crev-dev/cargo-crev#249]
I am unable to use `crev fetch` with `git@` (ssh) URLs.

```bash
> RUST_BACKTRACE=1 cargo crev fetch url git@github.com:dpc/crev-proofs.git
Fetching git@github.com:dpc/crev-proofs.git... Error: authentication required but no callback set; class=Ssh (23)
```

Using the same `fetch` command with an `https` URL works.

`crev` was installed using `cargo`; I am running on Mac OS X Mojave (10.14.6).

```
> cargo crev --version
cargo-crev-crev 0.9.0
```

**Comment by dpc** [comment]
In principle the proof repos should be open for public. And we fetch them using `libgit2` so handling authentication challenge might be tricky here.

Definitely low prio for me right now. "Just use https" for now. PRs always welcome though.

**Comment by dpc** [comment]
We could detect the link not be `http(s)` and warn the user...

**Comment by BatmanAoD** [comment]
I'm not sure that would have worked in my case, because I had a git config
setting to replace https calls with git@ (for some reason that I don't
remember). That wouldn't be detectable until after the git lib had done its
thing, would it?

**Comment by dpc** [comment]
Oh dear. I don't know. And it's a minor enough case, that I'll just ignore it for now. :D

**Comment by pickfire** [comment]
Yup, it affected me as well. I can't take fetch my own review. Because in my git config I did.
```
[url "git@github.com:pickfire"]
        insteadOf = https://github.com/pickfire
```
Is there any way to just force https in cargo crev?

**Comment by dpc** [comment]
I doubt there's one generic way to force it. We could however (probably) register a callback or something and at very least print some useful info - like a link to a Wiki page. If anyone is feeling brave - please take a stab at writing a PR, otherwise we might get to it at some point. :)

**Comment by pickfire** [comment]
The issue is probably due to libgit2 https://github.com/rust-lang/git2-rs/issues/600

git clone works with the same url though. Maybe once `gitoxide` crate can do cloning we can use that as alternative.

**Comment by eirnym** [comment]
cargo supports git auth via ssh as well

Also there's libraries to support such auth, one of them is [auth_git2](https://lib.rs/crates/auth-git2)

**Comment by eirnym** [comment]
@dpc I'd like to present real world use case, why `authentication` provided by library above (`auth_git2`) could be required, and when attitude "we support only public repositories" doesn't work.

I agree, that security information in general should be open. Let's imagine (quite common) situation:

I work in a company, which has company-wide private repositories. Many teams produce company-private creates and publish into a private company-wide repository, which is a proxy to `crates.io`.

Per company policy, authenticated access is the must to company-wide repositories, it can be done using SSO for browser and SSH key for other applications.

To improve company-level security company will try to use tools such as `cargo-crev` to verify their tools. To use `cargo-crev` in current state, company must broke their security policy, when you could see a repository when you have permissions to, as database publishes some internal information.

**Comment by dpc** [comment]
I'm not against supporting it, but it is just not a priority. If anyone submits a PR, probably current maintainer will be happy to merge.
